boards/launchxl/layout.ld

```diff
@@ -0,0 +1,261 @@
+/*
```

Comment:
Does it possibly make more sense to just integrate the CCFG sections into the generic linker script (with a `CCFG_LENGTH = DEFINED(CCFG_LENGTH) ? CCFG_LENGTH : 0;` guard so as not to affect other boards)?

The effective diff here is really just
```diff
>   ccfg (rx) : ORIGIN = CCFG_ORIGIN, LENGTH =  CCFG_LENGTH
...
>     .ccfg : {
>         KEEP(*(.ccfg))
>     } > ccfg
```

I don't think it's necessary to duplicate the whole linker file just for that. I strongly prefer having fewer things that need updating if/when things change.

Reply:
Personally I'd be much more comfortable to add optionals to the generic linker script if there were more than one board which used this in the main repo. Premature generalisation can get out of hand rather quickly. I'd prefer to explicitly declare that the launchxl board requires a different linker script (even if the diff is minimal) until more boards use the same functionality.

We were thinking about separating the ccfg and the launchxl, since there is no need to flash the ccfg everytime you update the kernel; but I think that'd be an entirely new PR. Unsure what the best practice in this case would be though (python script in tools/ ?)

Reply:
Seems like while we are thinking about what the diff actually is is a good time to add the generic functionality (particularly if it is small like Pat indicates). The more time that goes by the harder it is likely to be to reconcile the two.

Reply:
I guess when it's this small, it makes sense to include it